### ISSUE TITLE
4.0.0beta2/qt5: further misadventures in Qt and SQL

### ISSUE THREAD

**Issue by kernelpanic3** [ElvishArtisan/rivendell#734]
Hello again!

While working on a brand-new testbed for the campus station I volunteer at to evaluate a potential future move towards Rivendell, I've run into the Qt bug from #673 again, and managed to consistently trigger it this time.

This time it cropped up while editing groups, and is reproducible by 1. adding a new group in RDAdmin, and 2. setting any range of default cart numbers. I've tried several different combinations ranging from 1-2 to 600001-700000, and they all resulted in the same issue as before:
```
ASSERT failure in QList<T>::at: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 544
Aborted
```
Curiously, the bug does _not_ happen when _editing_ a group - just when creating a new one.

Very similar setup as before - 4.0.0beta2, Ubuntu 20.04 LTS, MySQL 8.0.26, however this time I opted to use the dedicated appliance script instead of compiling myself.

-----

I also ran into a previously unseen SQL issue when attempting to change one of the default groups from all-caps to mixed case ... the group was instead wiped from the DB, with a deluge of errors like so:
```
QSqlQuery::value: not positioned on a valid record
for query: select `DESCRIPTION` from `GROUPS` where `NAME`="Macros"

QSqlQuery::value: not positioned on a valid record
for query: select `DEFAULT_TITLE` from `GROUPS` where `NAME`="Macros"

QSqlQuery::value: not positioned on a valid record
for query: select `NOTIFY_EMAIL_ADDRESS` from `GROUPS` where `NAME`="Macros"
```
and so on, one for every value related to the group.

As this is a long-term project, this testbed setup is here to stay, so I'm happy to poke things with a stick if needed.

Thanks a lot in advance ... very excited by the progress being made! :)

**Comment by ElvishArtisan** [comment]
> While working on a brand-new testbed for the campus station I volunteer at to evaluate a potential future move towards Rivendell, I've run into the Qt bug from #673 again, and managed to consistently trigger it this time.
>
> This time it cropped up while editing groups, and is reproducible by 1. adding a new group in RDAdmin, and 2. setting any range of default cart numbers. I've tried several different combinations ranging from 1-2 to 600001-700000, and they all resulted in the same issue as before:
>
> ASSERT failure in QList<T>::at: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 544
Aborted

Unable to reproduce here thus far. [This is on CentOS 7 -- I will try Ubuntu as well].

**Comment by ElvishArtisan** [comment]
> I also ran into a previously unseen SQL issue when attempting to change one of the default groups from all-caps to mixed case ... the group was instead wiped from the DB, with a deluge of errors like so:

Confirmed here.

**Comment by ElvishArtisan** [comment]
> I also ran into a previously unseen SQL issue when attempting to change one of the default groups from all-caps to mixed case ... the group was instead wiped from the DB, with a deluge of errors like so:

Fixed in 1d485f3. Please test!
